cli/args/config_file.rs

```diff
@@ -345,11 +349,57 @@ impl FilesConfig {
   }
 }
 
-#[derive(Clone, Debug, Default, Deserialize)]
+/// Choose between flat and nested files configuration.
+///
+/// `files` has precedence over `deprecated_files`.
+/// when `deprecated_files` is present, a warning is logged.
+///
+/// caveat: due to default values, it's not possible to distinguish between
+/// an empty configuration and a configuration with default values.
+/// `{ "files": {} }` is equivalent to `{ "files": { "include": [], "exclude": [] } }`
+/// and it wouldn't be able to emit warning for `{ "files": {}, "exclude": [] }`.
+///
+/// # Arguments
+///
+/// * `files` - Flat configuration.
+/// * `deprecated_files` - Nested configuration. ("Files")
+fn choose_files(
+  files: SerializedFilesConfig,
+  deprecated_files: SerializedFilesConfig,
+) -> SerializedFilesConfig {
+  const DEPRECATED_FILES: &str =
+    "Warning: \"files\" configuration is deprecated";
+  const FLAT_CONFIG: &str = "`include` and `exclude`";
+
+  let (files_nonempty, deprecated_files_nonempty) =
+    (!files.is_empty(), !deprecated_files.is_empty());
+
+  match (files_nonempty, deprecated_files_nonempty) {
+    (true, true) => {
+      log::warn!("{DEPRECATED_FILES} and ignored by {FLAT_CONFIG}.");
```

Comment:
[windows test](https://github.com/denoland/deno/actions/runs/4431832625/jobs/7776591675) is failing because `log::warn` output is polluting lint output. should i remove this line?

Reply:
thanks for the insight. fixed output and deno.jsonc format through [d4bda2d](https://github.com/denoland/deno/pull/17799/commits/d4bda2d33de1cfe37129fb1ad01655bc7e68465c)..[96b9016](https://github.com/denoland/deno/pull/17799/commits/96b9016fc9256ffab1e54cab70270ed7097c13c0)
deno.jsonc in `test/` directory are kept out to check if legacy usage warning is generated.